src/test/optimizer/strategy/dependent_group_by_reduction_rule_test.cpp

```diff
@@ -0,0 +1,324 @@
+    table_c->add_soft_unique_constraint({ColumnID{0}, ColumnID{2}}, IsPrimaryKey::Yes);
+    storage_manager.add_table("table_c", table_c);
+    stored_table_node_c = StoredTableNode::make("table_c");
+    column_c_0 = stored_table_node_c->get_column("column0");
+    column_c_1 = stored_table_node_c->get_column("column1");
+    column_c_2 = stored_table_node_c->get_column("column2");
+
+    table_d = std::make_shared<Table>(TableColumnDefinitions{{"column0", DataType::Int, false}}, TableType::Data, 2,
+                                      UseMvcc::Yes);
+    storage_manager.add_table("table_d", table_d);
+    stored_table_node_d = StoredTableNode::make("table_d");
+
+    table_e = std::make_shared<Table>(column_definitions, TableType::Data, 2, UseMvcc::Yes);
+    table_e->add_soft_unique_constraint({ColumnID{0}, ColumnID{1}}, IsPrimaryKey::Yes);
+    table_e->add_soft_unique_constraint({ColumnID{2}}, IsPrimaryKey::No);
+    storage_manager.add_table("table_e", table_e);
+    stored_table_node_e = StoredTableNode::make("table_e");
+    column_e_0 = stored_table_node_e->get_column("column0");
+    column_e_1 = stored_table_node_e->get_column("column1");
+    column_e_2 = stored_table_node_e->get_column("column2");
+
+    rule = std::make_shared<DependentGroupByReductionRule>();
+  }
+
+  std::shared_ptr<DependentGroupByReductionRule> rule;
+
+  std::shared_ptr<Table> table_a, table_b, table_c, table_d, table_e;
+  std::shared_ptr<StoredTableNode> stored_table_node_a, stored_table_node_b, stored_table_node_c, stored_table_node_d,
+      stored_table_node_e;
+  LQPColumnReference column_a_0, column_a_1, column_a_2;
+  LQPColumnReference column_b_0, column_b_1, column_b_2;
+  LQPColumnReference column_c_0, column_c_1, column_c_2;
+  LQPColumnReference column_d_0;
+  LQPColumnReference column_e_0, column_e_1, column_e_2;
+};
+
+// Test simple cases
+TEST_F(DependentGroupByReductionRuleTest, SimpleCases) {
+  // Early out for LQP without any aggregates
+  // clang-format off
+  {
+    auto lqp =
+    PredicateNode::make(equals_(column_a_0, 17), stored_table_node_a);
+
+    const auto unmodified_lqp = apply_rule(rule, lqp);
+    EXPECT_EQ(lqp, unmodified_lqp);
+  }
+
+  // Early out for LQP where table does not have a primary key/unique constraint
+  {
+    auto lqp =
+    PredicateNode::make(equals_(column_d_0, 17), stored_table_node_d);
+
+    const auto unmodified_lqp = apply_rule(rule, lqp);
+    EXPECT_EQ(lqp, unmodified_lqp);
+  }
+  // clang-format on
+}
+
+// Test that a removable column is removed when the single primary key column is present.
+// Check for the restored colum order.
+TEST_F(DependentGroupByReductionRuleTest, SingleKeyReduction) {
+  // clang-format off
+  auto lqp =
+  AggregateNode::make(expression_vector(column_a_0, column_a_1),
+                      expression_vector(sum_(column_a_0), sum_(column_a_1),
+                                        sum_(column_a_2)),
```

Comment:
Formatting: Nodes should take a single line, the next node should be indented. (Applies to whole file)

Reply:
Even though is breaks the line length limit we usually use?

Reply:
Added `  // NOLINT` as done in other tests.

Reply:
> Even though is breaks the line length limit we usually use?

Yes